java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -767,4 +791,48 @@ public String getClientLibToken() {
   public String getQuotaProjectId() {
     return quotaProjectId;
   }
+
+  /**
+   * Returns the resolved endpoint for the Service to connect to Google Cloud
+   *
+   * <p>The resolved endpoint is always in `host:port` format
+   */
+  @InternalApi
+  public String getResolvedHost(String serviceName) {
+    if (universeDomain != null && universeDomain.isEmpty()) {
+      throw new IllegalArgumentException("Universe Domain cannot be empty");
```

Comment:
I felt it's wrong a method that "Returns the resolved endpoint for the Service to connect to Google Cloud" works only when `universeDomain` is set.

Reply:
This is a required check. If universedomain is set as `""`, we must error out.

Reply:
I misunderstood the meaning of the code. Added suggested edits.

Reply:
Then how about setting the requirement in `setUniverseDomain`?

Reply:
I think that makes sense. Guidance for temporary Apiary is to use whatever is passed in as the universe domain. The empty check is for handwritten and gapics. I'll ask about the empty string check, but it shouldn't be blocking for this.